scripts/check-lessons.sh

```
@@ -0,0 +1,29 @@
+LESSONS_DIR="$ROOT/docs/lessons"
+INDEX="$LESSONS_DIR/_index.md"
+[ -d "$LESSONS_DIR" ] || { echo "check-lessons: docs/lessons/ not present, nothing to check"; exit 0; }
+rc=0
+dupes="$(ls "$LESSONS_DIR" | grep -oE '^lesson-[0-9]+' | sort | uniq -d || true)"
+if [ -n "$dupes" ]; then
+  echo "check-lessons: lesson numbers used more than once:"
+  for n in $dupes; do ls "$LESSONS_DIR" | grep -E "^${n}-" | sed 's/^/  /'; done
+  rc=1
+fi
+if [ -f "$INDEX" ]; then
+  for f in $(ls "$LESSONS_DIR" | grep -E '^lesson-[0-9]+.*\.md$'); do
```

Shellcheck (0.11.0)

[warning] Don't use ls | grep. Use a glob or a for loop with a condition to allow non-alphanumeric filenames.

(SC2010)